extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts

```diff
@@ -89,8 +89,8 @@ function generateMarkdownForInterpretedResult(interpretedResult: AnalysisAlert,
       ...generateMarkdownForCodeSnippet(codeSnippet, language, highlightedRegion),
     );
   }
-  const alertMessage = buildMarkdownAlertMessage(interpretedResult);
-  lines.push(alertMessage);
+  const alertMessageLines = buildMarkdownAlertMessage(interpretedResult, language);
```

Comment:
At first I found it a little confusing that the code paths stuff is shown as part of the "message" - they are 2 different things, it just happens that the UX bundles them together - you get an expandable message if there are code paths to show.

This is not necessarily a good suggestion or one that should be implemented, but I wonder if the code would be simpler/more readable if we first build some markdown for code paths (if there are any) and then hand it to the `buildMarkdownAlertMessage` that knows whether to build a normal message or an expandable section based on that.

It might be worth trying that out. Or not! See what you think 😸

Reply:
I've (hopefully!) tidied this up a bit: The alert message is always rendered the same, but if there are paths we show an expandable "Show paths" section ([preview](https://github.com/github/vscode-codeql/blob/shati-patel/sharing-path-results/extensions/ql-vscode/test/pure-tests/remote-queries/markdown-generation/data/interpreted-results/path-problem/results-repo1.md))